src/main/webapp/app/entities/result/result.component.ts

```diff
@@ -121,6 +201,28 @@ export class ResultComponent implements OnInit, OnChanges {
         return this.result!.hasFeedback;
     }
 
+    hasParticipationResults(): boolean {
+        if (!this.participation) {
+            return false;
+        }
+        return this.participation.results && this.participation.results.length > 0;
+    }
+
+    isSubmissionInDueTime(): boolean {
+        if (this.participation.submissions[0] && this.participation.submissions[0].submissionDate && this.participation.exercise.dueDate) {
```

Comment:
Please, use `const submission = this.participation.submissions[0]` to avoid getting every time first submission from participation